src/modules/robot/Robot.cpp

```diff
@@ -194,58 +208,54 @@ void Robot::on_config_reload(void *argument)
     this->max_speeds[Z_AXIS]  = THEKERNEL->config->value(z_axis_max_speed_checksum    )->by_default(  300.0F)->as_number() / 60.0F;
 
-    Pin alpha_step_pin;
-    Pin alpha_dir_pin;
-    Pin alpha_en_pin;
-    Pin beta_step_pin;
-    Pin beta_dir_pin;
-    Pin beta_en_pin;
-    Pin gamma_step_pin;
-    Pin gamma_dir_pin;
-    Pin gamma_en_pin;
-
-    alpha_step_pin.from_string( THEKERNEL->config->value(alpha_step_pin_checksum )->by_default("2.0"  )->as_string())->as_output();
-    alpha_dir_pin.from_string(  THEKERNEL->config->value(alpha_dir_pin_checksum  )->by_default("0.5"  )->as_string())->as_output();
-    alpha_en_pin.from_string(   THEKERNEL->config->value(alpha_en_pin_checksum   )->by_default("0.4"  )->as_string())->as_output();
-    beta_step_pin.from_string(  THEKERNEL->config->value(beta_step_pin_checksum  )->by_default("2.1"  )->as_string())->as_output();
-    beta_dir_pin.from_string(   THEKERNEL->config->value(beta_dir_pin_checksum   )->by_default("0.11" )->as_string())->as_output();
-    beta_en_pin.from_string(    THEKERNEL->config->value(beta_en_pin_checksum    )->by_default("0.10" )->as_string())->as_output();
-    gamma_step_pin.from_string( THEKERNEL->config->value(gamma_step_pin_checksum )->by_default("2.2"  )->as_string())->as_output();
-    gamma_dir_pin.from_string(  THEKERNEL->config->value(gamma_dir_pin_checksum  )->by_default("0.20" )->as_string())->as_output();
-    gamma_en_pin.from_string(   THEKERNEL->config->value(gamma_en_pin_checksum   )->by_default("0.19" )->as_string())->as_output();
-
-    float steps_per_mm[3] = {
-        THEKERNEL->config->value(alpha_steps_per_mm_checksum)->by_default(  80.0F)->as_number(),
-        THEKERNEL->config->value(beta_steps_per_mm_checksum )->by_default(  80.0F)->as_number(),
-        THEKERNEL->config->value(gamma_steps_per_mm_checksum)->by_default(2560.0F)->as_number(),
-    };
-
     // TODO: delete or detect old steppermotors
+#if 0 //TODO: Add unregistering to StepperMotor/StepTicker so we can delete old motors 
+    for (auto a : actuators)
+        delete a;
+#endif
+    actuators.clear();
+
     // Make our 3 StepperMotors
-    this->alpha_stepper_motor  = new StepperMotor(alpha_step_pin, alpha_dir_pin, alpha_en_pin);
-    this->beta_stepper_motor   = new StepperMotor(beta_step_pin,  beta_dir_pin,  beta_en_pin );
-    this->gamma_stepper_motor  = new StepperMotor(gamma_step_pin, gamma_dir_pin, gamma_en_pin);
+    struct
+    {
+        uint16_t pin_checksums[3];
+        uint16_t per_mm_checksum;
+        uint16_t per_sec_checksum;
+        char const *pin_defaults[3];
+        float per_mm_default;
+    } p[] =
+    {
+        { ACTUATOR_CHECKSUMS("alpha"),   {"2.0", "0.5",  "0.4" },   80.0F },
+        { ACTUATOR_CHECKSUMS("beta"),    {"2.1", "0.11", "0.10"},   80.0F },
+        { ACTUATOR_CHECKSUMS("gamma"),   {"2.2", "0.20", "0.19"}, 2560.0F },
+        { ACTUATOR_CHECKSUMS("delta"),   { "nc",   "nc",   "nc"},   80.0F },
```

Comment:
Well for current, there is a direct mapping between greek letter and on-board motor, so it doesn't matter if we are setting current for an actuator or an extruder.

This *does* make things a bit harder to understand for beginners, but that's a job for the documentation, not for the code.